### DISCUSSION TITLE
Long term plans for this project?

### DISCUSSION THREAD

**Discussion by bkushigian** [b-inary/postflop-solver#18] [General]
As I've mentioned before, I'm very excited about this project. I've been wanting a high-quality OS solver to show up, and preferably one in Rust :)

Anyway, just wondering what your long term plans are for this project. Are you trying to make something with a similar feature set as PIO (e.g., node locking, UPI, etc)? Have you thought at all about preflop solves? Or is this just for fun?

Thanks again for all your work!

**Comment by b-inary** [comment]
My current mid-term goal is to develop a free GTO solver that is practical enough without requiring the purchase of PioSOLVER or GTO+. I will not implement UPI because I have no interest in getting on the PioSOLVER's ecosystem, but I would like to implement node-locking (by the way, incorporating node-locking into the engine is probably easy, and I am struggling with what to do with the UI).

Of course, one of the main motivations is simply because the development is fun. I am not getting any monetary reward, and if I no longer enjoy developing, I will not be able to continue it, even if my goals are not being met.

To be honest, it is vague about the long-term goals. As for solving preflop, I am skeptical about the usefulness of the solver, in part because it requires a large amount of abstraction in practice. If abstraction is needed anyway, I think it makes more sense to use deep learning. My underlying idea is to provide open-source tools that help reveal poker, and this project is based on that idea, but I am not envisioning the best path.

**Comment by bkushigian** [comment]
Mid term: that's great, I think this has so much value added for the community. One of the reasons that new players don't use solvers is that the entry price is so high. Removing tooling friction is immensely valuable IMO.

Re: node locking: what's tricky about the UI? Are you referring to the GUI, or the node locking API you want to offer from the solver? I'd imagine the GUI bit is the trickiest part about this entire thing.

Re: UPI: interesting! what about UPI do you dislike? Are you opposed to a functionally equivalent interface? I guess the beauty of OSS is that I can implement UPI (or something similar) and offer that as an external lib. But scriptability is very valuable, and I'd imagine there is a lot of existing tooling using UPI that could plug your solver in instead of PIO, and that could add to adoptability. Anyway, this might be a thing I take on at some point in an external lib...

Re: Preflop: interesting, I haven't looked into deep-learning based preflop sims (i'd imagine something like poker snowie?) but that would be really cool! I've been meaning to learn more about DL/ML for a while so I'd be open to brushing off my math skills.

**Comment by b-inary** [comment]
Node locking: Yes, UI meant GUI.

UPI: It is not that I dislike UPI, but it is just not a priority for me, as it seems to require a lot of tedious work to accommodate UPI. Since this is open source, users have full access to the solver and can write their scripts (although it requires a development environment and learning of Rust).

Preflop: I don't know the technical details of Snowie, but I think a deep network that learns counterfactual values, like DeepStack, is open to possibilities. However, this approach requires significant computational resources for learning and is certainly hard to employ in hobbyist development.
